catalog/spatial/registry/registry-identification-plugin/src/main/java/org/codice/ddf/registry/identification/IdentificationPlugin.java

```diff
@@ -91,10 +119,80 @@ public CreateRequest process(CreateRequest input)
     @Override
     public UpdateRequest process(UpdateRequest input)
             throws PluginExecutionException, StopProcessingException {
+
+        OperationTransaction operationTransaction = (OperationTransaction) input.getProperties()
+                .get(Constants.OPERATION_TRANSACTION_KEY);
+
+        if (operationTransaction == null) {
+            throw new UnsupportedOperationException(
+                    "Unable to get OperationTransaction from UpdateRequest");
+        }
+
+        if (Requests.isLocal(input)) {
+            for (Map.Entry entry : input.getUpdates()) {
+                Metacard updateMetacard = (Metacard) entry.getValue();
+                if (updateMetacard != null) {
+
+                    for (Metacard metacard : operationTransaction.getPreviousStateMetacards()) {
```

Comment:
Might want to make a map of the operationTransaction metacards based on registry id outside the top level loop so you only have to iterate through it once. You could then just pull it out of the map here.